userdetails-plugin/grails-app/controllers/au/org/ala/userdetails/UserDetailsWebServicesInterceptor.groovy

```diff
@@ -30,13 +30,19 @@ class UserDetailsWebServicesInterceptor {
     }
 
     boolean before() {
-        if (!authorisedSystemService.isAuthorisedRequest(request, response, null, 'users/read')) {
-            log.warn("Denying access to $actionName from remote addr: ${request.remoteAddr}, remote host: ${request.remoteHost}")
-            response.sendError(HttpStatus.SC_UNAUTHORIZED)
-
+        try {
+            if (!authorisedSystemService.isAuthorisedRequest(request, response, null, 'users/read')) {
+                log.warn("Denying access to $actionName from remote addr: ${request.remoteAddr}, remote host: ${request.remoteHost}")
+                response.sendError(HttpStatus.SC_UNAUTHORIZED)
+
+                return false
+            }
+            return true
+        }
+        catch (Exception e){
+            response.sendError(HttpStatus.SC_UNAUTHORIZED, e.getMessage())
```

Comment:
IMHO Should probably log the exception instead of sending the exception message in the http response body

Reply:
Fixed